circuit_knitting/cutting/cut_finding/best_first_search.py

```diff
@@ -268,7 +269,7 @@ def optimization_pass(
 
             self.update_minimum_reached(cost)
 
-            if cost is None or self.cost_bounds_exceeded(cost):
+            if cost is None or self.cost_bounds_exceeded(cost):  # pragma: no cover
```

Comment:
Do you understand why this change resulted in this branch losing coverage?

Reply:
I think in exploring the multiple minima, it was encountering a ``None`` state for one of the test cases. I'd have to look into this more to figure out exactly what was going on.

Reply:
If I remember correctly, when exploring multiple minima, the search ends when a `None` state is encountered. That doesn't happen when `stop_at_first_main` is set to `False`. I can check this early next week.

Reply:
I have added a test for this case (here: [c4c2968](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/pull/585/commits/c4c29685807ecf9f6709a2933c3c564446ecf2df)). The test also allows you to describe the workflow of the optimizer at a more granular level. We can force `BestFirstSearch.optimize()` to run until it encounters a `None` state which then tests this line. I am a little concerned, though, that I overdid this one a little bit and maybe excluding this line from coverage would have been okay instead.

Reply:
I think it's a good test, thanks